### PR TITLE
code refactoring for container package

### PR DESCRIPTION
**Pull request by artemgurzhii**

### CODE REVIEW

**Comment by stefanpenner** [comment]
CI seems unhappy

**Review by stefanpenner — APPROVED** [review]
:shipit: once green.

**Comment by artemgurzhii** [comment]
@homu done!

File: packages/container/lib/container.js [hunk 1]
```diff
@@ -330,8 +330,7 @@ function lookup(container, fullName, options = {}) {
 }
 
 function isSingletonClass(container, fullName, { instantiate, singleton }) {
-  return (singleton !== false && isSingleton(container, fullName)) &&
-         (instantiate === false && !shouldInstantiate(container, fullName));
+  return singleton !== false && isSingleton(container, fullName) && !instantiate && !shouldInstantiate(container, fullName);
 }
 
 function isSingletonInstance(container, fullName, { instantiate, singleton }) {
```
Comment (stefanpenner) [inline packages/container/lib/container.js]: We typically want to avoid lines that are too long. I also feel the former is easier to read

Comment (artemgurzhii) [inline packages/container/lib/container.js]: I was looking at this [pull request](https://github.com/emberjs/ember.js/pull/15014/files). Should I replace it?

File: packages/container/lib/registry.js [hunk 2]
```diff
@@ -732,31 +732,31 @@ Registry.prototype = {
...
+      injections.push(...this.fallback.getInjections(fullName));
     }
     return injections;
   },
 
   getTypeInjections(type) {
     let injections = this._typeInjections[type] || [];
     if (this.fallback) {
-      injections = injections.concat(this.fallback.getTypeInjections(type));
+      injections.push(...this.fallback.getTypeInjections(type));
```
Comment (bekzod) [inline packages/container/lib/registry.js]: aren you modifying `this._typeInjections[type] ` here ?

File: packages/container/lib/registry.js [hunk 3]
```diff
@@ -732,31 +732,31 @@ Registry.prototype = {
   getInjections(fullName) {
     let injections = this._injections[fullName] || [];
     if (this.fallback) {
-      injections = injections.concat(this.fallback.getInjections(fullName));
+      injections.push(...this.fallback.getInjections(fullName));
```
Comment (bekzod) [inline packages/container/lib/registry.js]: so you are modifying `this._injections[fullName]`, because former did not do that

Comment (bekzod) [inline packages/container/lib/registry.js]: I hope you are not polluting `this._injections[fullName] ` here :P

**Comment by bekzod** [comment]
@stefanpenner is merging master to PR is right way to update with master ?

**Comment by stefanpenner** [comment]
@bekzod the best way is to rebase.

```sh
git fetch origin
# while on your working branch
git rebase origin/master
```

**Comment by bekzod** [comment]
@stefanpenner just asking because this one was merged, but if it is accepted then cool

File: packages/container/lib/registry.js (hunk 3 above)
Comment (bekzod) [inline packages/container/lib/registry.js]: before/after is not same here, so each time you call `getInjections` it it modifies `this._injections[fullName]` if there is fallback.  am I wrong ? @stefanpenner @artemgurzhii

**Comment by locks** [comment]
Closing this one due to conflicts, feel free to give it another go!

**Pull request closed without merging**